src/invidious/routes/images.cr

```diff
@@ -149,16 +223,44 @@ module Invidious::Routes::Images
     id = env.params.url["id"]
     name = env.params.url["name"]
 
-    headers = HTTP::Headers{":authority" => "i.ytimg.com"}
+    headers = (
+      {% unless flag?(:disable_quic) %}
+        if CONFIG.use_quic
+          HTTP::Headers{":authority" => "i.ytimg.com"}
+        else
+          HTTP::Headers.new
+        end
+      {% else %}
+        HTTP::Headers.new
+      {% end %}
+    )
 
     if name == "maxres.jpg"
       build_thumbnails(id).each do |thumb|
-        if YT_POOL.client &.head("/vi/#{id}/#{thumb[:url]}.jpg", headers).status_code == 200
-          name = thumb[:url] + ".jpg"
-          break
-        end
+        thumbnail_resource_path = "/vi/#{id}/#{thumb[:url]}.jpg"
+        # Logic here is short enough that manually typing them out should be fine.
+        {% unless flag?(:disable_quic) %}
+          if CONFIG.use_quic
+            if YT_POOL.client &.head(thumbnail_resource_path, headers).status_code == 200
+              name = thumb[:url] + ".jpg"
+              break
+            end
+          else
+            if HTTP::Client.head("https://i.ytimg.com#{thumbnail_resource_path}").status_code == 200
+              name = thumb[:url] + ".jpg"
+              break
+            end
+          end
```

Comment:
I'm wondering something: why isn't the `if CONFIG.use_quic` check made at the `YT_POOL` pool level, rather than here?